index.src.html

```diff
@@ -332,15 +460,64 @@
     <a>control messages</a> in the <a>control message queue</a>, and all pending
     callbacks.
 
-    When invoked, run the <a>Reset</a> algorithm.
+    When invoked, run the <a>Reset VideoDecoder</a> algorithm.
   </dd>
 
   <dt><dfn method for=VideoDecoder>close()</df></dt>
   <dd>
-    Immediately aborts all pending work and releases system resources. Close is
-    permanent.
+    Immediately aborts all pending work and releases <a>system resources</a>.
+    Close is permanent.
 
-    When invoked, run the <a>Close</a> algorithm.
+    When invoked, run the <a>Close VideoDecoder</a> algorithm.
+  </dd>
+</dl>
+
+Algorithms {#videodecoder-algorithms}
+-------------------------------------
+<dl>
+  <dt><dfn>Output VideoFrame</dfn> (with |output|)</dt>
+  <dd>
+    Run these steps:
+    1. Let |planes| be a sequence of {{Plane}}s containing the decoded video
+        frame data from |output|.
+    2. Let |pixelFormat| be the {{PixelFormat}} of |planes|.
+    3. Let |frameInit| be a {{VideoFrameInit}} with the following keys:
+        1. Let {{VideoFrameInit/timestamp}} and {{VideoFrameInit/duration}} be
+            the {{EncodedVideoChunk/timestamp}} and
+            {{EncodedVideoChunk/duration}} from the {{EncodedVideoChunk}}
+            associated with |output|.
+        2. Let {{VideoFrameInit/codedWidth}} and {{VideoFrameInit/codedHeight}}
+            be the width and height of the decoded video frame |output| in
+            pixels, prior to any cropping or aspect ratio adjustments.
```

Comment:
I think the `video` width and height map to what I'm calling `displayWidth` and `dipslayHeight`. There are 3 different types of sizing for video frames

- coded size (how many pixels you need to allocate memory for). codedWidth corresponds to "line stride" as described in this excellent post: https://medium.com/@oleg.shipitko/what-does-stride-mean-in-image-processing-bba158a72bcd
- crop size (the region of coded size you actually want to display)
- display size (the final number of pixels you want to see on the screen. this may imply stretching the pixels from the crop size)